test/scales/modifiedLogScaleTests.ts

```diff
@@ -2,187 +2,241 @@
+      beforeEach(() => {
+        scale = new Plottable.Scales.ModifiedLog(base);
+        scale.padProportion(0);
+      });
 
-      let maxInMiddle = 0;
-      scale.domainMax(maxInMiddle);
-      assert.deepEqual(scale.domain(), [requestedDomain[0], maxInMiddle], "upper end was set even if requested value cuts off some data");
+      it("expands single value domains to [value / base, value * base].sort()", () => {
+        let singleValue = 15;
+        scale.addIncludedValuesProvider((scale: Plottable.Scales.ModifiedLog) => [singleValue]);
+        assert.deepEqual(scale.domain(), [singleValue / base, singleValue * base],
+          "positive single-value extent was expanded to [value / base, value * base]");
+        singleValue = -15;
+        scale.autoDomain();
+        assert.deepEqual(scale.domain(), [singleValue * base, singleValue / base],
+          "negative single-value extent was expanded to [value * base, value / base]");
+        singleValue = 0;
+        scale.autoDomain();
+        assert.deepEqual(scale.domain(), [-base, base],
+          "zero single-value extent was expanded to [base, -base]");
+      });
 
-      scale.autoDomain();
-      assert.deepEqual(scale.domain(), requestedDomain, "calling autoDomain() overrides domainMax()");
+      it("can force the minimum of the domain with domainMin()", () => {
+        let requestedDomain = [-5, 5];
+        scale.addIncludedValuesProvider((scale) => requestedDomain);
+
+        let minBelowBottom = -10;
+        assert.strictEqual(scale.domainMin(minBelowBottom), scale, "the scale is returned by the setter");
+        assert.strictEqual(scale.domainMin(), minBelowBottom, "can get the domainMin()");
+        assert.deepEqual(scale.domain(), [minBelowBottom, requestedDomain[1]], "lower end of domain was set by domainMin()");
+
+        let minInMiddle = 0;
+        scale.domainMin(minInMiddle);
+        assert.deepEqual(scale.domain(), [minInMiddle, requestedDomain[1]],
+          "lower end was set even if requested value cuts off some data");
+
+        scale.autoDomain();
+        assert.deepEqual(scale.domain(), requestedDomain, "calling autoDomain() overrides domainMin()");
+        assert.strictEqual(scale.domainMin(), scale.domain()[0], "returns autoDomain()-ed min value after autoDomain()-ing");
+
+        let minEqualTop = scale.domain()[1];
+        scale.domainMin(minEqualTop);
+        assert.deepEqual(scale.domain(), [minEqualTop, minEqualTop * base],
+          "domain is set to [min, min * base] if the requested value is >= autoDomain()-ed max value");
+
+        scale.domainMin(minInMiddle);
+        let requestedDomain2 = [-10, 10];
+        scale.addIncludedValuesProvider((scale) => requestedDomain2);
+        assert.deepEqual(scale.domain(), [minInMiddle, requestedDomain2[1]], "adding another ExtentsProvider doesn't change domainMin()");
+      });
```

Comment:
That is probably an excellent idea, as I've seen this code so many times already. But I started bottom-up, and no idea what happens on the QuantitativeScale level. Will make a note of it